pkg/provider/provider.go

```diff
@@ -669,3 +514,16 @@ func validateObjectFormat(objectFormat, objectType string) error {
 	}
 	return nil
 }
+
+// getObjectVersion parses the id to retrieve the version
+// of object fetched
+func getObjectVersion(id string) string {
+	splitID := strings.Split(id, "/")
+	return splitID[len(splitID)-1]
```

Comment:
Is this logic always true? Where is this documented?

Reply:
Opened an issue - https://github.com/Azure/azure-rest-api-specs/issues/10825